cpp/log/log_lookup-inl.h

```diff
@@ -68,8 +69,7 @@ void LogLookup<Logged>::UpdateFromSTH(const ct::SignedTreeHead& sth) {
     // handled more gracefully. E.g. we could retry a failed update
     // a number of times -- but until we know under which conditions
     // the database might fail (database busy?), just die.
-    CHECK_EQ(Database<Logged>::LOOKUP_OK,
-             db_->LookupByIndex(sequence_number, &logged))
+    CHECK(it->GetNextEntry(&logged))
```

Comment:
This used to get the entry by index, using `sequence_number`, so the `CHECK_EQ` at line 77 was just a sanity check ("I asked for 10 and got 11?!?")... But now, you might have a gap, and get some other entry index...

That said, it was a `LookupByIndex()` wrapped in a `CHECK_EQ`, so it was already going to crash (I guess that it's a precondition for this method that the database has the entries we need for the STH we get passed in?), so it might be okay (or, at least, no worse)?